mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster.go

```diff
@@ -2557,6 +2589,9 @@ func flattenInstanceGroupConfig(d *schema.ResourceData, icg *dataproc.InstanceGr
 		data["num_instances"] = icg.NumInstances
 		data["machine_type"] = tpgresource.GetResourceNameFromSelfLink(icg.MachineTypeUri)
 		data["min_cpu_platform"] = icg.MinCpuPlatform
+		if icg.MinNumInstances != 0 {
```

Comment:
Rather than just guard on the value being nonzero, we should pass through the type of config we're checking and write the value conditionally. If the API began to allow a minimum number of primary instances and started passing through a value to a user (a backwards compatible change), the user would get an error on their next Terraform run as we tried to write to a value that didn't exist.